pkg/client/transport/transport.go

```diff
@@ -55,9 +55,6 @@ func TLSConfigFor(c *Config) (*tls.Config, error) {
 	if !(c.HasCA() || c.HasCertAuth() || c.TLS.Insecure) {
 		return nil, nil
 	}
-	if c.HasCA() && c.TLS.Insecure {
```

Comment:
Ohh, that seems way better! When I first started writing the patch I though something was setting it by error, but didn't found when it was written and I assumed maybe it was expected.  But what you say makes way more sense, will do that! :)